common/ipmi/ipmb.c

```diff
@@ -131,7 +115,7 @@ void insert_node(ipmi_msg_cfg *pnode, ipmi_msg *msg, uint8_t index)
 {
   ipmi_msg_cfg *ptr_start = pnode;
 
-  osMutexAcquire(mutex_id[index], osWaitForever);
+  k_mutex_lock(&mutex_id[index], K_FOREVER);
```

Comment:
For function insert node, find node and timeout handler, in timeout case, it risks a condition that BIC sending IPMB success but fail to record IPMB request package, which makes BIC take IPMB response as invalid package according to IPMI specification.
To avoid thread pending forever in mutex, we think we can add mutex status check. If BIC lock mutex fail due to one second timeout, will return directly and record a mutex lock fail in BIC console.
Is that make sense to you? Or do you think there is a better solution?

Reply:
The main issue is to not hang forever.  Having a one second timeout and recording the lock failure is sufficient to meet this requirement of not hanging the BMC if the mutex is unable to be acquired.

Reply:
OK, we will fix it soon.